Virtual appointment function updates to beta

benmicrosoft opened the pull request:
This PR is to add information for our new Virtual appointment function that is replacing our existing resource and crud methods. This update also includes the Deprecate notice

srikhs commented on api-reference/beta/api/onlinemeeting-getvirtualappoinmentjoinweburl.md:
````diff
@@ -0,0 +1,109 @@
+}
+-->
+``` http
+GET https://graph.microsoft.com/beta/me/onlineMeeting/MSpkYzE3Njc0Yy04MWQ5LTRhZGItYmZi/getVirtualAppointmentJoinWebURL
````
`onlineMeeting` -> `onlineMeetings`

benmicrosoft commented on api-reference/beta/api/onlinemeeting-getvirtualappoinmentjoinweburl.md:
```diff
@@ -0,0 +1,109 @@
+description: "Function to return virtualAppointment join link."
+author: "benmicrosoft"
+ms.localizationpriority: medium
+ms.prod: "cloud-communications"
```
test

srikhs commented on api-reference/beta/api/onlinemeeting-getvirtualappoinmentjoinweburl.md:
```diff
@@ -0,0 +1,109 @@
+
+| Permission type                        | Permissions (from least to most privileged)                                            |
+|:---------------------------------------|:---------------------------------------------------------------------------------------|
+| Delegated (work or school account)     | VirtualAppointment.Read, OnlineMeetings.Read                                  |
```
The ones with ReadWrite Permissions (VirtualAppointment.ReadWrite, OnlineMeetings.ReadWrite, VirtualAppointment.Read.All will also be able to use this GET endpoint since they also include the read part.

benmicrosoft commented:
Roadmap item https://microsoftgraph.visualstudio.com/Roadmap/_workitems/edit/16973

benmicrosoft commented:
API.md file - https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding/pullrequest/13555

Csdl - https://dev.azure.com/msazure/One/_git/AD-AggregatorService-Workloads/pullrequest/7887217

srikhs commented on api-reference/beta/api/virtualappointment-getvirtualappoinmentjoinweburl.md:
````diff
@@ -0,0 +1,105 @@
+}
+-->
+``` http
+GET /me/onlineMeetings/{onlineMeetingId}/getVirtualAppointmentJoinWebURL
````
this has to be `getVirtualAppointmentJoinWebUrl` to align with our changes in csdl

srikhs commented on api-reference/beta/api/virtualappointment-getvirtualappoinmentjoinweburl.md:
````diff
@@ -0,0 +1,105 @@
+-->
+``` http
+GET /me/onlineMeetings/{onlineMeetingId}/getVirtualAppointmentJoinWebUrl
+GET /users/{userId}/onlineMeetings/{onlineMeetingId}/getVirtualAppointmentJoinWebUurl
````
there is an additional u

dodaromike commented:
@benmicrosoft There are many invalid or incorrectly formatted links in these files that need to be fixed before the PR can be approved or merged to main.

benmicrosoft commented:
thanks @dodaromike. can you help us understand why we are getting the Invalid file link: '../api/virtualappointment-getvirtualappointmentjoinweburl.md' on all the pages? This link seems valid to the new .md file?

dodaromike commented:
> thanks @dodaromike. can you help us understand why we are getting the Invalid file link: '../api/virtualappointment-getvirtualappointmentjoinweburl.md' on all the pages? This link seems valid to the new .md file?

It looks like some of them are in the same directory as the file they're linked from; they don't need ../api part of the path.  Those in the resources directory appear to be correct.

dodaromike commented:
@benmicrosoft  I think you need changelog entries, at least for getVirtualAppointmentJoinWebUrl.   See https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211742/Changelog

dodaromike commented:
@benmicrosoft  Did you add the snippet file to your PR?
[api-reference/beta/api/virtualappointment-getvirtualappointmentjoinweburl.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/benmicrosoft-VAPreviewdocumenation/api-reference/beta/api/virtualappointment-getvirtualappointmentjoinweburl.md)
Line 76, Column 1: [Error: include-not-found] Invalid include link: '../includes/snippets/csharp/get-virtualappointment-csharp-snippets.md'.

dodaromike commented:
@benmicrosoft  I don't know what's wrong with this one:

benmicrosoft commented:
> @benmicrosoft Did you add the snippet file to your PR? [api-reference/beta/api/virtualappointment-getvirtualappointmentjoinweburl.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/benmicrosoft-VAPreviewdocumenation/api-reference/beta/api/virtualappointment-getvirtualappointmentjoinweburl.md) Line 76, Column 1: [Error: include-not-found] Invalid include link: '../includes/snippets/csharp/get-virtualappointment-csharp-snippets.md'.

I thought these code snippets auto generated? Either way, I added the snippet  header to the example request and deleted the existing snippets since they were pointing to the old get file. This cleared the error. https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211752/Generate-code-snippets

benmicrosoft commented:
> @benmicrosoft I don't know what's wrong with this one: [api-reference/beta/toc.yml](https://github.com/microsoftgraph/microsoft-graph-docs/blob/benmicrosoft-VAPreviewdocumenation/api-reference/beta/toc.yml) Line 17075, Column 13: [Error: yaml-syntax-error] While parsing a block mapping, did not find expected key.

@dodaromike I believe is has to do with the formatting of the indentation. We need the new GetwebURL file to be above the deprecated resource and methods... I'm not sure how to clear this either.

benmicrosoft commented:
Hey @dodaromike any update on this content review? This new schema is already published on beta and we are ready to open it to traffic. cc @srikhs

dodaromike commented:
@benmicrosoft  I think the TOC problem is a blocker.  @Lauragra What do you suggest?

benmicrosoft commented:
@dodaromike the TOC syntax error is now resolved. cc @srikhs

benmicrosoft commented:
@Lauragra In this PR we are deprecating our existing resource and CRUD methods, and replacing them with a new Get function for Virtual appointments. I just want to make sure this TOC looks correct. Thanks!

benmicrosoft commented:
Thanks @dodaromike! Can you change the PR label from content review in progress to content review complete. Thanks for all your hard work!

Lauragra commented on api-reference/beta/api/virtualappointment-delete.md:
```diff
@@ -13,6 +13,9 @@ Namespace: microsoft.graph
 
 [!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
 
+> [!CAUTION]
```
Please update the TOC entries for the deprecated methods to include (deprecated), for visibility. Also, how is it that we're adding a net-new method for virtual appointment if the resource itself is deprecated?

benmicrosoft replied:
Hey @Lauragra - We need to deprecate our current virtual appointment resource and CRUD methods because of a compliance issue with storage. After working with graph council, we have decided to push a new get function that returns our virtual appointment join link. This function does not need the resource to work. I'm wondering since this is net new functionality, if we should break the TOC out like this for better clarity: 

Get Virtual Appointment (preview)
-Get Virtual appointment join link

Virtual Appointment (deprecated)
-Create (deprecated)
-Read (deprecated)
-Update (deprecated)
-Delete (deprecated)

I'm open to other suggestions as well, I want to make this clear to developers

Lauragra replied:
Oh, I see your comment here now. Yes, I agree that we should surface the new function outside of the Virtual appointment node. It might make sense to name the node itself "Virtual appointment (preview)", to distinguish from the deprecated node, although it's tricky since there isn't any resource (yet). Is the new replacement resource going to still be called virtualAppointment, or something else?

Lauragra commented on api-reference/beta/api/virtualappointment-getvirtualappointmentjoinweburl.md:
```diff
@@ -0,0 +1,89 @@
+doc_type: apiPageType
+---
+
+# virtualAppointment: getVirtualAppointmentJoinWebUrl
```
Is the virtualAppointment resource deprecated, and if so, why are we adding this function? Does this method topic also need the deprecation note?

benmicrosoft replied:
@Lauragra This function is replacing the deprecated functionality.

Lauragra commented on api-reference/beta/toc.yml:
```diff
@@ -17074,6 +17074,8 @@ items:
           - name: Virtual appointment (preview)
```
In the TOC entries for the deprecated resource and methods, we should put (deprecated).

benmicrosoft replied:
Updated to reflect

Lauragra commented on api-reference/beta/toc.yml:
```diff
@@ -17074,6 +17074,8 @@ items:
           - name: Virtual appointment (preview)
             href: resources/virtualappointment.md
             items:
+            - name: Get Virtual appointment join link
```
I think that temporarily we might want to surface this function as a peer of the Virtual appointment node, so that it's discoverable and not missed by the fact that it's nested under a resource that's deprecated.

benmicrosoft replied:
Updated to reflect

Lauragra commented on api-reference/beta/api/virtualappointment-delete.md:
```diff
@@ -13,6 +13,9 @@ Namespace: microsoft.graph
 [!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
 
+> [!CAUTION]
+  > The Virtual appointment resource and supporting methods are deprecated and will stop returning data on May 31, 2023. Existing apps that use this feature should be updated to the new [Get Virtual appointment join link](../api/virtualappointment-getvirtualappointmentjoinweburl.md).
```
I suggest revising the second sentence of the deprecation note to the following:
"We recommend that you update existing apps that use this API to use the [Get Virtual appointment join link](../api/virtualappointment-getvirtualappointmentjoinweburl.md) function."

That function doesn't perform the same action as the CRUD operations on virtualAppointment though, right? So is the second sentence applicable in all the method topics?

benmicrosoft replied:
Good point, the function only really does create/get of the link. I updated with the above language on the put and get pages, and the original virtual appointment.md file. The other pages I just marked with the first sentence in the deprecation notice since the function isn't really a substitute.

Lauragra commented on api-reference/beta/api/virtualappointment-getvirtualappointmentjoinweburl.md:
```diff
@@ -0,0 +1,88 @@
+Get a join web URL for [Microsoft Virtual Appointments](/microsoft-365/frontline/virtual-appointments). This web URL includes enhanced business-to-customer experiences such as mobile broswer join and virtual lobby rooms. 
+
+> [!NOTE]
+> For the preview period, customers with access to virtual appointments with their Microsoft 365 licenses benefit from unlimited virtual appointment calls via the Microsoft Graph API. We'll provide additional details about licensing requirements before the preview period ends.
```
What do you mean by "preview period"? Is that something different than being in beta? Does this statement intend to imply that this API will incur fees when it goes to v1.0?

benmicrosoft replied:
I removed this phrase, the monetization effort with onlinemeeting is still being finalized for v1.0

Lauragra commented on api-reference/beta/api/virtualappointment-getvirtualappointmentjoinweburl.md:
```diff
@@ -0,0 +1,88 @@
+| Application                            | VirtualAppointment.Read.All, VirtualAppointment.ReadWrite.All
+
+> [!NOTE]
+> When this API is made generally available later this year, Virtual appointment will transition to requiring virtual appointment permissions and no longer accept online meeting permissions. This will give developers more granular control over virtual appointment permissions.
```
We don't make public commitments to future availability. I recommend that you remove this statement.

benmicrosoft replied:
removed

Lauragra commented on api-reference/beta/api/virtualappointment-getvirtualappointmentjoinweburl.md:
```diff
@@ -0,0 +1,88 @@
+
+### Response
+The following example shows the response.
+>**Note:** The response object shown here might be shortened for readability.
```
If only one property is returned, we can probably remove this statement, unless you think that you'll add more to this API in the future?

benmicrosoft replied:
removed
